core/vm/evm.go

```diff
@@ -452,7 +452,12 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 
 // Create creates a new contract using code as deployment code.
 func (evm *EVM) Create(caller ContractRef, code []byte, gas uint64, value *big.Int) (ret []byte, contractAddr common.Address, leftOverGas uint64, err error) {
-	contractAddr = crypto.CreateAddress(caller.Address(), evm.StateDB.GetNonce(caller.Address()))
+	_, callerIsContract := caller.(ContractStruct)
+	nonce := evm.StateDB.GetNonce(caller.Address())
+	if evm.ChainConfig().IsHawaii(evm.Context.BlockNumber) && !callerIsContract {
```

Comment:
evm codebase has 2 places calling `evm.Create`

1. https://github.com/iotexproject/go-ethereum/blob/master/core/state_transition.go#L209
the caller here is EOA, this is usual case of deploying a contract. In this case, increment the nonce after Hawaii height

2. https://github.com/iotexproject/go-ethereum/blob/master/core/vm/instructions.go#L702
the caller here is `Contract struct`, this is contract creating child-contract inside itself. In this case, we don't increment the nonce, keep the same behavior